multiwallet.go

```diff
@@ -429,6 +434,60 @@ func (mw *MultiWallet) LinkExistingWallet(walletName, walletDataDir, originalPub
 	})
 }
 
+// walletsReadCopy is concurrently-safe way to read from mw.wallets map.
+func (mw *MultiWallet) walletsReadCopy() map[int]*Wallet {
+	result := make(map[int]*Wallet, len(mw.wallets))
```

Comment:
Lock before reading `len(mw.wallets)`. Also applies to the badWalletsReadyCopy method.

Reply:
missed this one, good catch